mindsdb/integrations/handlers/weaviate_handler/weaviate_handler.py

```diff
@@ -0,0 +1,633 @@
+        tables = []
+        if query_tables:
+            tables = [table["class"] for table in query_tables["classes"]]
+        table_name = pd.DataFrame(
+            columns=["table_name"],
+            data=tables,
+        )
+        return Response(resp_type=RESPONSE_TYPE.TABLE, data_frame=table_name)
+
+    def get_columns(self, table_name: str) -> HandlerResponse:
+        table_name = table_name.capitalize()
+        # check if table exists
+        try:
+            table = self._client.schema.get(table_name)
+        except ValueError:
+            return Response(
+                resp_type=RESPONSE_TYPE.ERROR,
+                error_message=f"Table {table_name} does not exist!",
+            )
+        data = pd.DataFrame(
+            data=[
+                {"COLUMN_NAME": column["name"], "DATA_TYPE": column["dataType"][0]}
+                for column in table["properties"]
+            ]
+        )
+        return Response(data_frame=data, resp_type=RESPONSE_TYPE.OK)
+
+    def add_metadata(self, data: dict, table_name: str):
+        table_name = table_name.capitalize()
+        metadata_table_name = table_name.capitalize() + "_metadata"
+        self._client.schema.get(metadata_table_name)
+        # getting existing metadata fields
+        added_prop_list = [
+            prop["name"]
+            for prop in self._client.schema.get(metadata_table_name)["properties"]
+        ]
+        # as metadata columns are not fixed, at every entry, a check takes place for the columns
+        for prop in data.keys():
+            if prop not in added_prop_list:
+                if isinstance(data[prop], int):
+                    add_prop = {
+                        "name": prop,
+                        "dataType": ["int"],
+                    }
+                elif isinstance(data[prop][0], datetime):
+                    add_prop = {
+                        "name": prop,
+                        "dataType": ["date"],
+                    }
+                else:
+                    add_prop = {
+                        "name": prop,
+                        "dataType": ["string"],
+                    }
+                # when a new column is identified, it is added to the metadata table
+                self._client.schema.property.create(metadata_table_name, add_prop)
+        metadata_id = self._client.data_object.create(
+            data_object=data, class_name=table_name.capitalize() + "_metadata"
+        )
+        return metadata_id
+
+
+connection_args = OrderedDict(
+    weaviate_url={
+        "type": ARG_TYPE.STR,
+        "description": "weaviate url",
+        "required": True,
+    },
+    weaviate_api_key={
+        "type": ARG_TYPE.STR,
+        "description": "weaviate API KEY",
+        "required": True,
+    },
+)
+
+connection_args_example = OrderedDict(
+    weaviate_url="https://sample-s3x8blpm.weaviate.network",
+    weaviate_api_key="mdMdLfFSbwcWqLI6NUNUseIoIS8yxnY84U5C",
```

Comment:
let's not hardcode api key please

Reply:
resolved in e8944f0fe38ecffba0a915419bd098a57c562e55